### PR TITLE
Document _getUpdateComplete in LitElement docs

### PR DESCRIPTION
**Pull request by aomarks**
Fixes https://github.com/Polymer/lit-element/issues/746

### CODE REVIEW

File: docs/_guide/lifecycle.md [hunk 1]
```diff
@@ -279,6 +279,19 @@ The `updateComplete` Promise resolves when the element has finished updating. Us
   this.updateComplete.then(() => { /* do stuff */ });
   ```
 
+To await additional state before fulfilling the `updateComplete` promise, override the `_getUpdateComplete` method. For example, it may be useful to await the update of a child element here. First await `super._getUpdateComplete()`, then any subsequent state.
```
Comment (arthurevans) [inline docs/_guide/lifecycle.md]: If overriding `_getUpdateComplete` is less common than using `this.updateComplete`, I think this note should be _after_ the main example.

I'd suggest moving this after the `updateComplete` example, and introducing it with a L4 heading:

`#### Overriding updateComplete`

(to be clear, I'm suggesting inserting it just before `## Examples`, ~L302 in this PR.

File: docs/_guide/lifecycle.md [hunk 2]
```diff
@@ -279,6 +279,19 @@ The `updateComplete` Promise resolves when the element has finished updating. Us
...
+
+Note that it's recommended to override the `_getUpdateComplete` method instead of the `updateComplete` getter to ensure compatibility with users who are using TypeScript's ES5 output (see [TypeScript#338](https://github.com/microsoft/TypeScript/issues/338)).
```
Comment (arthurevans) [inline docs/_guide/lifecycle.md]: You can cut "Note that" here.

Comment (aomarks) [reply docs/_guide/lifecycle.md]: Done.

File: docs/_guide/lifecycle.md (hunk 1 above)
Comment (aomarks) [reply docs/_guide/lifecycle.md]: SGTM, done!

**Pull request merged**